apps/block_scout_web/lib/block_scout_web/controllers/api/rpc/address_controller.ex

```diff
@@ -569,7 +569,7 @@ defmodule BlockScoutWeb.API.RPC.AddressController do
 
   defp list_tokens(address_hash) do
     case Etherscan.list_tokens(address_hash) do
-      [] -> {:error, :not_found}
+      empty_list when is_list(empty_list) and empty_list == [] -> {:error, :not_found}
```

Comment:
can we revert this?

Reply:
dialyzer fails if I revert it: https://github.com/celo-org/blockscout/runs/7210742669?check_suite_focus=true

Reply:
that's the only reason why I refactored it

Reply:
this is super weird, in the first case the pattern is handled in the `else` clause, and the second is just nonsensical as it is the first clause in the list

```
lib/block_scout_web/controllers/api/rpc/address_controller.ex:204:pattern_match
The pattern can never match the type.
Pattern:
{:ok, _token_list}
Type:
{:error, :not_found}
________________________________________________________________________________
lib/block_scout_web/controllers/api/rpc/address_controller.ex:573:pattern_match_cov
The pattern
variable_token_list
can never match, because previous clauses completely cover the type
[].
```

Reply:
Well, initially I did it with `length(empty_list) == 0` and exactly then credo warning was indeed firing ;-) Going to add a dialyzer comment then.

Reply:
@rkachowski so I just in the end fixed it properly by adapting the typespec